src/components/auth/PrivateRoute.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import {useSelector} from 'react-redux';
+import {Route, Redirect, RouteProps} from 'react-router-dom';
+
+import {RootState} from '../../store';
+
+interface Props extends RouteProps {
```

Comment:
`type Props = RouteProps & {}` instead.

Also, name this as `PrivateRouteProps`. This helps when we have a type mismatching problem. We will be able to use its name to identify which is giving the errors, increasing the debugability.